internal/command/add.go

```diff
@@ -0,0 +1,330 @@
+				Provider: provider,
+				Module:   args.Addr.Module.Module(),
+			}
+		} else {
+			// lacking any configuration to query, we'll go with a default provider.
+			absProviderConfig = &addrs.AbsProviderConfig{
+				Provider: addrs.NewDefaultProvider(ip),
+			}
+			providerLocalName = ip
+		}
+	} else {
+		if module != nil {
+			providerLocalName = module.LocalNameForProvider(absProviderConfig.Provider)
+		} else {
+			providerLocalName = absProviderConfig.Provider.Type
+		}
+	}
+
+	localProviderConfig := addrs.LocalProviderConfig{
+		LocalName: providerLocalName,
+		Alias:     absProviderConfig.Alias,
+	}
+
+	// Get the schemas from the context
+	if _, exists := schemas.Providers[absProviderConfig.Provider]; !exists {
+		diags = diags.Append(tfdiags.Sourceless(
+			tfdiags.Error,
+			"Missing schema for provider",
+			fmt.Sprintf("No schema found for provider %s. Please verify that this provider exists in the configuration.", absProviderConfig.Provider.String()),
+		))
+		c.View.Diagnostics(diags)
+		return 1
+	}
+
+	// Get the schema for the resource
+	schema, schemaVersion := schemas.ResourceTypeConfig(absProviderConfig.Provider, rs.Mode, rs.Type)
+	if schema == nil {
+		diags = diags.Append(tfdiags.Sourceless(
+			tfdiags.Error,
+			"Missing resource schema from provider",
+			fmt.Sprintf("No resource schema found for %s.", rs.Type),
+		))
+		c.View.Diagnostics(diags)
+		return 1
+	}
+
+	stateVal := cty.NilVal
+	// Now that we have the schema, we can decode the previously-acquired resource state
+	if args.FromResourceAddr != nil {
+		ri := resource.Instance(args.FromResourceAddr.Resource.Key)
+		if ri.Current == nil {
+			diags = diags.Append(tfdiags.Sourceless(
+				tfdiags.Error,
+				"No state for resource",
+				fmt.Sprintf("There is no state found for the resource %s, so add cannot populate values.", rs.String()),
+			))
+			c.View.Diagnostics(diags)
+			return 1
+		}
+
+		rio, err := ri.Current.Decode(schema.ImpliedType())
+		if err != nil {
+			diags = diags.Append(tfdiags.Sourceless(
+				tfdiags.Error,
+				"Error decoding state",
+				fmt.Sprintf("Error decoding state for resource %s: %s", rs.String(), err.Error()),
+			))
+			c.View.Diagnostics(diags)
+			return 1
+		}
+
+		if ri.Current.SchemaVersion != schemaVersion {
+			diags = diags.Append(tfdiags.Sourceless(
+				tfdiags.Error,
+				"Schema version mismatch",
+				fmt.Sprintf("schema version %d for %s in state does not match version %d from the provider", ri.Current.SchemaVersion, rs.String(), schemaVersion),
+			))
+			c.View.Diagnostics(diags)
+			return 1
+		}
+
+		stateVal = rio.Value
+	}
+
+	diags = diags.Append(view.Resource(args.Addr, schema, localProviderConfig, stateVal))
+	if diags.HasErrors() {
+		c.View.Diagnostics(diags)
+		return 1
+	}
+
+	return 0
+}
+
+func (c *AddCommand) Help() string {
+	helpText := `
+Usage: terraform [global options] add [options] ADDRESS
+
+  Generates a blank resource template. With no additional flags,
```

Comment:
This is a total nit :grimacing: but since we're saying "options" everywhere else here (which is consistent with how we've been evolving our docs/usage output elsewhere), it'd be nice to be consistent here and say "additional _options_" instead of "additional _flags_".